src/components/mapping-dialog/MappingDialog.tsx

```diff
@@ -44,8 +41,14 @@ const MappingDialog: React.FC<MappingDialogProps> = ({
     const [connectionSuccess, setConnectionSuccess] = useState(true);
     const api = instance.getApi();
 
-    const element = elements.length === 1 ? _.find(rows, ["id", elements[0]]) : undefined;
-    const mappedId = _.get(mapping, [model.getCollectionName(), element?.id ?? "", "mappedId"]);
+    if (elements.length > 1)
+        throw new Error("Applying mapping to more than one item is not supported yet");
+
+    const element = _.find(rows, ["id", elements[0]]);
+    const type = element?.__mappingType__ ?? element?.__type__ ?? "";
+    const model = d2ModelFactory(api, type as keyof D2ModelSchemas);
```

Comment:
We should not render the dialog if type is not found in element.

Type should be provided as a prop.